Source/WebKit/WebKit.xcodeproj/project.pbxproj

```diff
@@ -19666,6 +19669,69 @@
 			shellPath = /bin/sh;
 			shellScript = "xcrun clang -E -P -w -target ${ARCHS%% *}-${LLVM_TARGET_TRIPLE_VENDOR}-${LLVM_TARGET_TRIPLE_OS_VERSION}${LLVM_TARGET_TRIPLE_SUFFIX} - < \"${SCRIPT_INPUT_FILE_0}\" > \"${SCRIPT_OUTPUT_FILE_0}\"\n";
 		};
+		DD725EBD2D8938A500E72544 /* Audit SPI use */ = {
+			isa = PBXShellScriptBuildPhase;
+			buildActionMask = 2147483647;
+			dependencyFile = "$(DERIVED_FILES_DIR)/api-analyzer.d";
+			files = (
+			);
+			inputFileListPaths = (
+			);
+			inputPaths = (
+				"$(WTF_BUILD_SCRIPTS_DIR)/analyze-spi.sh",
+			);
+			name = "Audit SPI use";
+			outputFileListPaths = (
+			);
+			outputPaths = (
+				"$(DERIVED_FILE_DIR)/analyze-spi.timestamp",
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+			shellPath = /bin/sh;
+			shellScript = "\"${SCRIPT_INPUT_FILE_0}\" \"${BUILT_PRODUCTS_DIR}/JavaScriptCore.framework/JavaScriptCore\"{,.partial.sdkdb} \"${BUILT_PRODUCTS_DIR}/libwebrtc.dylib\" \"${BUILT_PRODUCTS_DIR}/WebCore.framework/WebCore\"{,.partial.sdkdb} \"${BUILT_PRODUCTS_DIR}/WebGPU.framework/WebGPU\"{,.partial.sdkdb} \"${BUILT_PRODUCTS_DIR}/libWebKitSwift\"{.dylib,.partial.sdkdb} \"${BUILT_PRODUCTS_DIR}/WebKit.framework/WebKit.partial.sdkdb\"\n";
```

Comment:
I'm concerned about the maintainability of inlining these shell scripts which seems to mostly be used to expand file paths with shell expansion. Can this be a separate shell script?

Reply:
Yeah, I prefer the per-"framework" approach is taken for Inspector/WebDriver code generation. See `generate-inspector-protocol-bindings.py` and its invocations in script phases.